crowbar_framework/app/models/service_object.rb

```diff
@@ -953,7 +957,11 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
       delay, pre_cached_nodes = queue_proposal(inst, element_order, new_elements, deps)
-      return [202, "Queued: waiting for #{delay.join(", ")}"] unless delay.empty?
+      unless delay.empty?
+        # force not processing the queue further
+        in_queue = true
```

Comment:
@aspiers that's part of the "We move some code that must always be run before returning in the global ensure", yes :-)